src/main/java/com/xpanxion/java/assignments/student7/Worker7.java

```diff
@@ -1,8 +1,26 @@
 package com.xpanxion.java.assignments.student7;
 
+import com.xpanxion.java.assignments.DataAccess;
+
+import java.util.stream.Collectors;
+
 public class Worker7 {
 
     public void ex1() {
-        System.out.println("Ex. 1...");
+        var products = DataAccess.getProducts();
+        var departments = DataAccess.getDepartments();
+
+        var correctProducts = products.stream()
+                .map(p -> {
+                    if (p.getDepartmentId() == 1) {
```

Comment:
This is correct.  But instead of using if statements, could we use a `HashMap`?  The answer is yes.  :-)  Can you noodle on that for a bit?

Reply:
just saw this after my commit should i try hashmap or does the new commit work?

Reply:
Please try a hash map.